route: Support routes created by iproute command

cathay4t opened the pull request:
In order to persistent routes created by iproute, the `Route.CONFIG`
should also be reported by `nispor`, so that it could be merged and
send to backend for writing to profile.

To do that, we need:
 * NetworkManager don't report interface which is unmanaged or external
   managed.
 * Nispor will include static routes in `Route.CONFIG`.
 * Since we are verifying the kernel status, we should allow kernel
   appending additional routes besides routes in config. So do that,
   we hard coded the `Route.USE_DEFAULT_ROUTE_TABLE` as 254 and ignore
   route metric difference, so RouteEntry could used for `issubset()`
   check.

Integration test case included.

ffmancera commented on libnmstate/nm/route.py:
```diff
@@ -37,52 +36,6 @@
-    return routes
-
-
 def _get_per_route_table_id(nm_route, default_table_id):
     table = nm_route.get_attribute(NM.IP_ROUTE_ATTRIBUTE_TABLE)
     return int(table.get_uint32()) if table else default_table_id
```
This function is not used anymore, please remove it. The same for the function `_nm_route_to_route` and `_get_default_route_config`. Thanks!

cathay4t replied:
Thanks. Removed.

ffmancera reviewed (changes requested):
There is just some leftovers. Other than that it looks good. Thanks!

ffmancera reviewed (dismissed):
LGTM, thanks!

cathay4t committed:
route: Support routes created by iproute command

In order to persistent routes created by iproute, the `Route.CONFIG`
should also be reported by `nispor`, so that it could be merged and
send to backend for writing to profile.

To do that, we need:
 * NetworkManager don't report interface which is unmanaged or external
   managed.
 * Nispor will include static routes in `Route.CONFIG`.
 * Since we are verifying the kernel status, we should allow kernel
   appending additional routes besides routes in config. So do that,
   we hard coded the `Route.USE_DEFAULT_ROUTE_TABLE` as 254 and ignore
   route metric difference, so RouteEntry could used for `issubset()`
   check.

Integration test case included.

Signed-off-by: Gris Ge <fge@redhat.com>

ffmancera reviewed (approved):
LGTM, thanks!
